### PR TITLE
Glass bottles with reagents can be used for crafting, empty ones will be used as tools if possible

### LINKED ISSUE #76250: You can't craft anything that requires reagents with bottles as containers for it.
## Reproduction:
1. Get everything needed for crafting something with reagent needed for it.
2. Fill `/glass/bottle` with the needed reagent.
3. Don't be able to craft anything.
# Improvised Shotgun Shells
image
image
image
image
# Dragonsbreath Shell
image
image
image
Works this way exclusively for `/glass/bottle` i think?
With `/glass/drinkingglass/shotglass` as example i could craft what i wanted.
Its not just `weapon ammo` category, couldn't craft IED either.

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
Fixes #76250

The problem was a glass bottle has `tool_behaviour = rollingpin` and so it took priority in this if condition
https://github.com/tgstation/tgstation/blob/93d4b6d6cd3b160989960f9f88c27640ab06ba2d/code/datums/components/crafting/crafting.dm#L125-L134
before it could check if its a reagent container

Now the priority is inverted, if the glass bottle has reagents inside it then it is used as a reagent container else if it is empty then it used as a rolling pin(i.e. checks for its tool behaviours)

## Changelog
:cl:
fix: glass bottles with reagents can be used for crafting, empty glass bottles will be used as tools(e.g. empty glass bottle as rolling pin)
fix: glass bottle with welding fuel can be used for crafting improvised shotgun shells
/:cl:

### CODE REVIEW

**Comment by SyncIt21** [comment]
@Time-Green can i bother you some more :) ?

File: code/datums/components/crafting/crafting.dm [hunk 1]
```diff
@@ -122,23 +122,19 @@
...
-				.["other"][item.type] += 1
-			else
-				if(is_reagent_container(item))
-					var/obj/item/reagent_containers/container = item
-					if(container.is_drainable())
-						for(var/datum/reagent/reagent in container.reagents.reagent_list)
-							.["other"][reagent.type] += reagent.volume
+			else if(is_reagent_container(item) && item.is_drainable() && length(item.reagents.reagent_list)) //some container that has some reagents inside it that can be drained
+				var/obj/item/reagent_containers/container = item
+				for(var/datum/reagent/reagent in container.reagents.reagent_list)
```
Comment (Time-Green) [inline code/datums/components/crafting/crafting.dm:127]: as anything?

**Comment by Time-Green** [comment]
also its TWO DAAAAYYYYYS

**Comment by SyncIt21** [comment]
> also its TWO DAAAAYYYYYS

I know, but i saw the commit history and you merged 2 of my PR's 3 hours apart so i thought maybe you had some free time :D

File: code/datums/components/crafting/crafting.dm (hunk 1 above)
Comment (SyncIt21) [reply code/datums/components/crafting/crafting.dm:127]: is it safe?

Comment (SyncIt21) [reply code/datums/components/crafting/crafting.dm:127]: I trust you

**Comment by Time-Green** [comment]
> > also its TWO DAAAAYYYYYS
> 
> I know, but i saw the commit history and you merged 2 of my PR's so i thought maybe you had some free time :D

Never :)

**Pull request merged**